source/common/grpc/context_impl.cc

```diff
@@ -85,12 +85,13 @@ void ContextImpl::chargeUpstreamStat(const Upstream::ClusterInfo& cluster,
 absl::optional<ContextImpl::RequestStatNames>
 ContextImpl::resolveDynamicServiceAndMethod(const Http::HeaderEntry* path) {
   absl::optional<Common::RequestNames> request_names = Common::resolveServiceAndMethod(path);
+
   if (!request_names) {
     return {};
   }
 
-  Stats::Element service = Stats::DynamicName(request_names->service_);
-  Stats::Element method = Stats::DynamicName(request_names->method_);
+  Stats::Element service = Stats::DynamicSavedName(request_names->service_);
+  Stats::Element method = Stats::DynamicSavedName(request_names->method_);
```

Comment:
Nit: I think it's worth adding a comment here that these new stat names may outlive the values in `request_names_` thus need to be turned into copies.

Reply:
Added, would take a look, please?